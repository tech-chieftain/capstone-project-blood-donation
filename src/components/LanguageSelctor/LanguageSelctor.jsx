import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { PropTypes } from 'prop-types';
import Enlish from './images/english.svg';
import Arabic from './images/arabic.svg';

function LanguageSelctor({ menuClose }) {
  const [languageflag, setlanguageflag] = useState(false);
  const { i18n } = useTranslation();
  const onclick = () => {
    setlanguageflag(!languageflag);
    i18n.changeLanguage(languageflag ? 'en' : 'ar');
    if (menuClose) menuClose();
  };

  return (
    <div>
      {languageflag === false ? (
        <button type="button" className="flex items-center" onClick={onclick}>
          <img
            className="w-8 h-8 rounded-full object-cover object-center"
            src={Arabic}
            alt="Arabic"
          />
          <p className="mx-2 text-white">عربي</p>
        </button>
      ) : (
        <button type="button" className="flex items-center" onClick={onclick}>
          <img
            className="w-6 h-6 rounded-full object-cover object-center mt-2"
            src={Enlish}
            alt="Enlish"
          />
          <p className="mx-2 text-white">En</p>
        </button>
      )}
    </div>
  );
}
LanguageSelctor.propTypes = {
  menuClose: PropTypes.func,
};

export default LanguageSelctor;
