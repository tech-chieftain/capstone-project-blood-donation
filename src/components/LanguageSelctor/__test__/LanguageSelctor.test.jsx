import React from 'react';
import renderer from 'react-test-renderer';
import LanguageSelctor from '../LanguageSelctor';

// This is an example of a Snapshot. It will create a new snapshot which all future
// tests will use to compare to.
it('renders correctly', () => {
  const tree = renderer
    .create(
      <LanguageSelctor
        languageflag
        setlanguageflag={() => console.log('Donate')}
        menuClose={() => console.log('Donate')}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
