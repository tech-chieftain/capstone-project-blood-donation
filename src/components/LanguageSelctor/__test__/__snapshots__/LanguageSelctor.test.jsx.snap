// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <button
    className="flex items-center"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="Arabic"
      className="w-8 h-8 rounded-full object-cover object-center"
      src="arabic.svg"
    />
    <p
      className="mx-2 text-white"
    >
      عربي
    </p>
  </button>
</div>
`;
